### ISSUE TITLE
bug(nn): [covariate_nn] + IOV (kappa) — warm-started FOCEI objective sits 7k–15k below the cold re-evaluation; fits never approach the analytic IOV model, 27× slower per eval

### ISSUE THREAD

**Issue by roninsightrx** [FeRx-NLME/ferx-core#1327]
## Summary

A `[covariate_nn]` model with IOV (`kappa` + `iov_column`) does not fit under FOCEI: the objective the outer optimizer sees during the run is **7 000–15 000 units below** what a cold-start evaluation of the same population parameters gives, so the optimizer chases a phantom, terminates after a handful of evals, and the reported fit is far worse than the analytic model it replaces. The same model without IOV, and the analytic model with IOV, are both fine. Each objective evaluation is also ~27× slower than the IOV analytic twin.

Found while running the DCM evaluation for #1310 on `ferx-testdata/busulfan_dominic` (1518 subjects, ~8 obs/subject, 2-cpt IV infusion, IOV on CL and V1 over up to 7 occasions).

## Reproduction (minutes)

`ferx-testdata/busulfan_dominic/ferx/dcm/dcm_h2_iov_s600.ferx` on `data/train_s600.csv` (600 subjects) — a 4→2→4 network, 22 weights, so it is small enough to run in a few minutes at 4 threads. Build with `--features ferx-core/nn` at or after `caba67d3`.

```
[parameters]
  block_omega (ETA_CL, ETA_V1) = [0.068, 0.038, 0.038]
  omega ETA_V2 ~ 0.077
  kappa KAPPA_CL ~ 0.0111
  kappa KAPPA_V1 ~ 0.0153
  sigma PROP_ERR ~ 0.05 (sd)
  sigma ADD_ERR  ~ 15 (sd)

[covariate_nn TYPICAL_PK]
  inputs = [LAGE, LWT, LHT, SEX]   center = [2.639, 3.727, 4.91, 0]   scale = [1.548, 0.907, 0.342, 1]
  outputs = [CL, V1, Q, V2]        layers = [2]   activation = tanh   output = softplus
  init = [11.6, 46.5, 14.3, 10.8]

[individual_parameters]
  CL = TYPICAL_PK.CL * exp(ETA_CL + KAPPA_CL)
  V1 = TYPICAL_PK.V1 * exp(ETA_V1 + KAPPA_V1)
  Q  = TYPICAL_PK.Q
  V2 = TYPICAL_PK.V2 * exp(ETA_V2)

[structural_model]
  pk two_cpt_iv(cl=CL, v1=V1, q=Q, v2=V2)
[error_model]
  DV ~ combined(PROP_ERR, ADD_ERR)
[fit_options]
  method = focei   iov_column = OCC   covariance = false
```

## What happens

| run | outer gradient | last optimizer eval | reported final (cold inner) | evals | wall |
|---|---|---|---|---|---|
| analytic base with IOV (`base_s600.ferx`) | analytic | 56 915.80 | 56 915.80 | 58 | ~5 min |
| DCM h2 + IOV | analytic (Dual2) | 71 456.17 | **80 766.57** | 386 | 13 min |
| DCM h2 + IOV, `gradient_method = fd` (BOBYQA) | FD | 72 181.92 ("Restored best-seen point") | **86 864.73** | 178 | 12 min |
| DCM h8 + IOV | analytic | 73 407.72 | **80 354.45** | 15 | 25 min (600 subj) |
| DCM h8 + IOV, full train (1215 subj) | analytic | 112 635 | **113 690.88** | 11 | 25 min |
| DCM h8, **no** IOV, same data | analytic | 59 838.63 | 59 838.63 | 2561 | 4 min |

Two facts separate the causes:

- **Cold evaluation is self-consistent.** `dcm_holdout` (an `outer_maxiter = 0` re-evaluation from the `.fitrx`, EBEs cold-started) on the training data returns 80 766.568430 for the analytic-gradient run and 56 915.799399 for the base — the reported numbers to the last digit. So the number the optimizer optimizes is the one that is wrong: the warm-started inner loop reports an objective ~9 300 too low for a network model with κ slots.
- **It is not the outer gradient.** Under FD/BOBYQA the trajectory and the cold value disagree by 14 700, more than under the analytic gradient.

And it is not a bad local mode of the same objective: the DCM+IOV runs never get anywhere near the analytic IOV base (best trajectory value 71 456 vs 56 916, holdout 31 915 vs 26 454), whereas the same DCM without IOV beats the no-IOV base on holdout by 100+.

## Where to look

`src/sens/provider.rs` — `iov_eta_only_derivs_dyn` builds the IOV inner η-gradient for a `[covariate_nn]` model with `ModelNnGuard` lifting the network outputs as constants, and returns `CombinedDerivs` with `dtheta` / `d2eta` **zero-filled** ("must not be consumed — they are not the derivatives, they are absent"). If the FOCEI objective's per-subject Hessian term on the IOV path consumes `d2eta` (or the mixed block) for the η-part of `[η_bsv, κ₁..κ_K]`, the objective is evaluated with a wrong (zero-curvature) Laplacian and comes out too low — which is the sign of the discrepancy observed. A second candidate is the warm-start bookkeeping for κ slots: the cold final pass is right and the warm path is wrong, so whatever state the warm path carries per occasion (cached NN outputs at a stale θ, or κ vectors keyed by a slot layout that the NN model builds differently) is worth checking. `Occasion slot 7: KAPPA_CL NaN` appears in the shrinkage report of both the DCM and the base, so that line is not the discriminator.

Also: each objective eval on the IOV DCM costs ~19 s on 600 subjects vs 0.7 s for the same network without IOV (and ~5 s for the analytic base with IOV). The `[covariate_nn]` weights are `n_theta` for the IOV walk's dual width even though the network output is η-independent, the same shape as #1300 on the TV-cov walk.

## What a fix needs to show

Trajectory value == cold re-evaluation at the final point (as the analytic IOV base already does, to 1e-6), and the DCM+IOV fit landing at or below the analytic IOV base on training OFV, since the network can represent the allometric model it replaces. `dcm_holdout` (`scratchpad` crate; 70 lines against `prepare_run` + `load_fit` + `fit` with `outer_maxiter = 0`) is the check.

Related: #1300 (TV-cov analytic walk for NN weights), #1310 (DCM tuning heuristics, where this dataset is a calibration point), #1317 (checkpoint stores latest not best).

**Comment by roninsightrx** [comment]
Verified on `44d2b380` against the busulfan_dominic repro in the issue: the 76-weight DCM + IOV on 600 subjects now runs analytic gradients at ~0.4 s/eval (was 19 s), and on the full 1215-subject split four DCM+IOV arms converge with optimizer last-eval == cold re-evaluation to 1e-5 (was 9 300 apart), Ω/κ/σ matching the analytic IOV base, and holdout OFV 83–115 below it. Numbers in https://github.com/FeRx-NLME/ferx-core/issues/1310#issuecomment-5632983400. The 600-subject subset still shows the block-Ω runaway of #1328, which is unrelated to this fix.
